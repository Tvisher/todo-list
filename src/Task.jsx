import React from 'react';
import classNames from 'classnames';


const Task = ({ task, onChange, onDelete }) => {
    const listItemClasses = classNames('list-item', { "list-item_done": task.done });

    return (
        <li className={listItemClasses}>
            <input type="checkbox"
                className='list-item__checkbox'
                defaultChecked={task.done}
                onChange={() => onChange(task.id)} />
            <p className='list-item__text'>{task.text}</p>
            <button onClick={() => onDelete(task.id)} className='list-item__delete'>+</button>
        </li>)
}

export default Task;