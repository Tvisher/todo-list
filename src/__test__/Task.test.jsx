import React from 'react';
import { shallow } from 'enzyme';
import Task from '../Task';

describe('<Task/>', () => {
    it('Shoud display done task', () => {
        const props = {
            task: {
                id: 1,
                done: true,
                text: 'Test text'
            },
            onChange: jest.fn(),
            onDelete: jest.fn()
        }
        const wparredComponent = shallow(<Task {...props} />);
        expect(wparredComponent).toMatchSnapshot();
    })

    it('Shoud display undone task', () => {
        const props = {
            task: {
                id: 1,
                done: false,
                text: 'Test text'
            },
            onChange: jest.fn(),
            onDelete: jest.fn()
        }
        const wparredComponent = shallow(<Task {...props} />);
        expect(wparredComponent).toMatchSnapshot();
    })

    it('Shoud update task on checkbox checked', () => {
        const props = {
            task: {
                id: 1,
                done: false,
                text: 'Test text'
            },
            onChange: jest.fn(),
            onDelete: jest.fn()
        }
        const wparredComponent = shallow(<Task {...props} />);
        wparredComponent.find('.list-item__checkbox').simulate('change')
        expect(props.onChange).toBeCalledWith(1);
    })

    it('Shoud delete task', () => {
        const props = {
            task: {
                id: 1,
                done: false,
                text: 'Test text'
            },
            onChange: jest.fn(),
            onDelete: jest.fn()
        }
        const wparredComponent = shallow(<Task {...props} />);
        wparredComponent.find('.list-item__delete').simulate('click')
        expect(props.onDelete).toBeCalledWith(1);
    })

})