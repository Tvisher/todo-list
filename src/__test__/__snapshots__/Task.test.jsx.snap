// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Task/> Shoud display done task 1`] = `
<li
  className="list-item list-item_done"
>
  <input
    className="list-item__checkbox"
    defaultChecked={true}
    onChange={[Function]}
    type="checkbox"
  />
  <p
    className="list-item__text"
  >
    Test text
  </p>
  <button
    className="list-item__delete"
    onClick={[Function]}
  >
    +
  </button>
</li>
`;

exports[`<Task/> Shoud display undone task 1`] = `
<li
  className="list-item"
>
  <input
    className="list-item__checkbox"
    defaultChecked={false}
    onChange={[Function]}
    type="checkbox"
  />
  <p
    className="list-item__text"
  >
    Test text
  </p>
  <button
    className="list-item__delete"
    onClick={[Function]}
  >
    +
  </button>
</li>
`;
